uplink/src/lib.rs

```diff
@@ -473,3 +492,54 @@ impl Uplink {
         self.shutdown_rx.recv_async().await
     }
 }
+
+pub struct BuiltIns {
+    tunshell_client: TunshellClient,
+    file_downloader: Option<FileDownloader>,
+    ota_installer: Option<OTAInstaller>,
+    script_runner: Option<ScriptRunner>,
+    #[cfg(target_os = "linux")]
+    logger: Option<JournalCtl>,
+    #[cfg(target_os = "android")]
+    logger: Option<Logcat>,
+    stat_collector: Option<StatCollector>,
+    process_handler: Option<ProcessHandler>,
+}
+
+impl BuiltIns {
+    pub fn spawn(self) {
+        thread::spawn(move || self.tunshell_client.start());
+
+        if let Some(file_downloader) = self.file_downloader {
+            thread::spawn(move || file_downloader.start());
+        }
+        if let Some(ota_installer) = self.ota_installer {
+            thread::spawn(move || ota_installer.start());
+        }
+        if let Some(logger) = self.logger {
+            thread::spawn(move || logger.start());
+        }
+        if let Some(stat_collector) = self.stat_collector {
+            thread::spawn(move || stat_collector.start());
+        }
+        if let Some(process_handler) = self.process_handler {
+            thread::spawn(move || process_handler.start());
+        }
+        if let Some(script_runner) = self.script_runner {
+            thread::spawn(move || {
+                let rt = tokio::runtime::Builder::new_current_thread()
+                    .thread_name("script_runner")
+                    .enable_io()
+                    .enable_time()
+                    .build()
+                    .unwrap();
+
+                rt.block_on(async move {
+                    if let Err(e) = script_runner.start().await {
```

Comment:
Declare the function with tokio proc marco like other apps instead of manually initializing the runtime

Reply:
If we need thread names, manually init a thread for every app in a different PR